model-optimizer/extensions/ops/non_max_suppression.py

```diff
@@ -31,13 +31,11 @@ def __init__(self, graph: Graph, attrs: dict):
         mandatory_props = {
             'type': self.op,
             'op': self.op,
-            'version': 'opset4',
+            'version': 'opset5',
             'infer': self.infer,
             'output_type': np.int64,
             'center_point_box': 0,
             'box_encoding': 'corner',
-            'in_ports_count': 5,
```

Comment:
Why did you remove these two lines? They are needed to properly initialize the number input/output ports for the operation.

Reply:
`in_ports_count` was added again. As for `out_ports_count`, NMS-1, NMS-3, NMS-4 have only 1 output port, but NMS-5 has 3 output ports. Moreover, if `out_ports_count == 3`, then we get an error
```sh
File "...\openvino\model-optimizer\mo\main.py", line 298, in main
    ret_code = driver(argv)
  File "...\openvino\model-optimizer\mo\main.py", line 265, in driver
    ret_res = emit_ir(prepare_ir(argv), argv)
  File "...\openvino\model-optimizer\mo\main.py", line 249, in emit_ir
    meta_info=get_meta_info(argv))
  File "...\openvino\model-optimizer\mo\pipeline\common.py", line 208, in prepare_emit_ir
    type_infer(graph)
  File "...\openvino\model-optimizer\mo\middle\passes\infer.py", line 272, in type_infer
    [port.get_data_type() for port in node.out_ports().values()]))
  File "...\openvino\model-optimizer\mo\middle\passes\infer.py", line 272, in <listcomp>
    [port.get_data_type() for port in node.out_ports().values()]))
  File "...\openvino\model-optimizer\mo\graph\port.py", line 366, in get_data_type
    self.node.name))
mo.utils.error.Error: The data type for out port 2 of node non_max_suppression_with_scores/NonMaxSuppressionV5 is not defined

Process finished with exit code 1
``` 
for the model EfficientDet-D7, and an error
```sh
File "...\openvino\model-optimizer\mo\main.py", line 298, in main
    ret_code = driver(argv)
  File "...\openvino\model-optimizer\mo\main.py", line 265, in driver
    ret_res = emit_ir(prepare_ir(argv), argv)
  File "...\openvino\model-optimizer\mo\main.py", line 249, in emit_ir
    meta_info=get_meta_info(argv))
  File "...\openvino\model-optimizer\mo\pipeline\common.py", line 208, in prepare_emit_ir
    type_infer(graph)
  File "...\openvino\model-optimizer\mo\middle\passes\infer.py", line 272, in type_infer
    [port.get_data_type() for port in node.out_ports().values()]))
  File "...\openvino\model-optimizer\mo\middle\passes\infer.py", line 272, in <listcomp>
    [port.get_data_type() for port in node.out_ports().values()]))
  File "...\openvino\model-optimizer\mo\graph\port.py", line 366, in get_data_type
    self.node.name))
mo.utils.error.Error: The data type for out port 1 of node 1162 is not defined

Process finished with exit code 1
```
for [Mask R CNN model](https://onnxzoo.blob.core.windows.net/models/opset_10/mask_rcnn/mask_rcnn_R_50_FPN_1x.onnx) (this model has NMS-4, not NMS-5).

Finally, if `out_ports_count == 1`, then it is incorrect for NMS-5, because NMS-5 has 3 outputs, and, e.g., 2 outputs of 3 are used in the EfficiendDet-D7 model.

Reply:
You can set different number of output port for different versions of the NMS operation in the operation constructor.

Reply:
Then we will have an abovementioned error with the TF model EfficientDet-D7, because this model has NMS-5, but uses only 2 of 3 outputs.